core/src/toga/app.py

```diff
@@ -554,7 +554,7 @@ def _startup(self):
         # This is a wrapper around the user's startup method that performs any
         # post-setup validation.
         self.startup()
-        self._verify_startup()
+        # self._verify_startup()
```

Comment:
Flagging as stray content.

Reply:
Stray content is still here.

Reply:
I am not sure if the verification should be skipped as I have not tested the requirement of `main_window` on other platforms. I have only tested this on windows. Can you also please confirm that the first test app in the PR description works on macos? If it works, then we can remove the `main_window` verification.